### PR TITLE
Unify drawer

### PR DESCRIPTION
**Pull request by Juveniel**
targets: https://github.com/telerik/kendo-themes-private/issues/216

### CODE REVIEW

File: packages/html/src/drawer/drawer-item.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    hasChildren?: boolean;
+    level?: number | string;
+};
+
+export type KendoDrawerItemState = { [K in (typeof states)[number]]?: boolean };
+
+export const DrawerItem = (
+    props: KendoDrawerItemProps &
+        KendoDrawerItemState &
+        React.HTMLAttributes<HTMLDivElement>
```
Comment (inikolova) [inline packages/html/src/drawer/drawer-item.spec.tsx:34]: Should be  React.HTMLAttributes<HTMLLIElement>

**Review by JoomFX — COMMENTED** [review]
I might be missing something, but I do not see the "states" in the visual tests.
Can we have them back?

**Comment by Juveniel** [comment]
> I might be missing something, but I do not see the "states" in the visual tests.
> 
> Can we have them back?

I have added a separate states test, so that we can fully reuse the newly created templates.

File: packages/bootstrap/scss/drawer/_variables.scss [hunk 2]
```diff
@@ -1,40 +1,105 @@
...
+/// The text color of the Drawer.
+/// @group drawer
 $kendo-drawer-text: $kendo-component-text !default;
+/// The border color of the Drawer.
+/// @group drawer
 $kendo-drawer-border: $kendo-component-border !default;
+/// The border width of the Drawer.
+/// @group drawer
 $kendo-drawer-border-width: 1px !default;
+/// The font family color of the Drawer.
```
Comment (inikolova) [inline packages/bootstrap/scss/drawer/_variables.scss:15]: Should be /// The font family of the Drawer.

File: packages/bootstrap/scss/drawer/_variables.scss [hunk 3]
```diff
@@ -1,40 +1,105 @@
...
+/// @group drawer
 $kendo-drawer-item-padding-y: $kendo-padding-md-y !default;
+/// The font size of the Drawer item.
+/// @group drawer
 $kendo-drawer-item-font-size: 16px !default;
+/// The line height of the Drawer item.
+/// @group drawer
 $kendo-drawer-item-line-height: $kendo-line-height-lg !default;
 
+/// Horizontal padding of the Drawer item in each level.
```
Comment (inikolova) [inline packages/bootstrap/scss/drawer/_variables.scss:61]: /// The horizontal padding...

File: packages/bootstrap/scss/drawer/_variables.scss [hunk 4]
```diff
@@ -1,40 +1,105 @@
...
+/// @group drawer
 $kendo-drawer-content-padding-y: $kendo-padding-md-y !default;
 
+/// The width of the Drawer scrollbar.
+/// @group drawer
 $kendo-drawer-scrollbar-width: 7px !default;
+/// The color of the Drawer scrollbar track.
+/// @group drawer
 $kendo-drawer-scrollbar-color: rgba( 156, 156, 156, .7 ) !default;
+/// The color of the Drawer scrollbar thumb.
```
Comment (inikolova) [inline packages/bootstrap/scss/drawer/_variables.scss:38]: ///The background color of the ...

**Comment by inikolova** [comment]
Overall LGTM. One note -> The drawer-item spec suggests usage of `<span className="k-spacer"></span>` for hierarchical items - I think we should note this in the needed changes for Angular (currently only jQuery suite renders it)

File: packages/html/src/drawer/templates/drawer-normal.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { Drawer, DrawerContent, DrawerItem, DrawerItemSeparator } from "..";
+
+export const DrawerNormal = (props) => (
+    <Drawer expanded
```
Comment (inikolova) [inline packages/html/src/drawer/templates/drawer-normal.tsx:4]: In my opinion DrawerNormal should not be expanded. I would suggest to add an additional template for expanded drawer

**Comment by JoomFX** [comment]
> > I might be missing something, but I do not see the "states" in the visual tests.
> > Can we have them back?
> 
> I have added a separate states test, so that we can fully reuse the newly created templates.

Yep, I missed it.
But now when I saw it, the "Selected + Hovered" state is not present there. We might want to add it.

**Review by JoomFX — APPROVED** [review]

**Review by inikolova — APPROVED** [review]

**Pull request merged**